Reference/conformance/conformance.pb.swift

```diff
@@ -147,44 +147,34 @@ struct Conformance_ConformanceRequest: SwiftProtobuf.Message, SwiftProtobuf.Prot
   ]
 
 
-  enum OneOf_Payload: ExpressibleByNilLiteral, SwiftProtobuf.OneofEnum {
+  enum OneOf_Payload: SwiftProtobuf.OneofEnum {
     case protobufPayload(Data)
     case jsonPayload(String)
-    case None
 
     static func ==(lhs: Conformance_ConformanceRequest.OneOf_Payload, rhs: Conformance_ConformanceRequest.OneOf_Payload) -> Bool {
       switch (lhs, rhs) {
       case (.protobufPayload(let l), .protobufPayload(let r)): return l == r
       case (.jsonPayload(let l), .jsonPayload(let r)): return l == r
-      case (.None, .None): return true
       default: return false
       }
     }
 
-    init(nilLiteral: ()) {
-      self = .None
-    }
-
-    init() {
-      self = .None
-    }
-
-    mutating func decodeField<T: SwiftProtobuf.Decoder>(decoder: inout T, fieldNumber: Int) throws {
-      if self != .None {
```

Comment:
`oneof`s are only decoded in one place, so "every calling place" is just moving it from the `oneof`'s decode method to the message's decode method. So, there should be no noticeable code size difference.

Performance-wise, our harness doesn't exercise `oneof`s right now. :(  Given the rarity of `oneof`s compared to other field types, any performance impact is probably negligible, and a reasonable price to pay for the right API.